src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -16,8 +16,8 @@ internal sealed class CompletionHelper
         private static readonly CompletionHelper CaseInsensitiveInstance = new CompletionHelper(isCaseSensitive: false);
 
         private readonly object _gate = new object();
-        private readonly Dictionary<(CultureInfo, string), PatternMatcher> _patternMatcherMap =
-             new Dictionary<(CultureInfo, string), PatternMatcher>();
+        private readonly Dictionary<(CultureInfo, string, bool includeMatchedSpans), PatternMatcher> _patternMatcherMap =
```

Comment:
Yes.  That is by design.  During typing, we see the same patterns appear over and over again.  So we just keep these around.  In practice, we've seen this use around 5MB of memory in practice in the large case.  But it benefits typing performance greatly by having so much data cached.